Add microbiome analysis workflows (#182)
galaxyproject/iwc: workflows/microbiome/README.md

# Microbiome Workflows

The following workflows can be used directly for microbiome data analysis, pathogen detection and tracking purposes. The workflows can be also adapted to any other sequencing technique. 

> To learn more about the following workflows and try them with real datasets, please check our tutorial on the Galaxy Training Network [GTN](https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)

Reviewer: Maybe put the Microbiome topic instead here

## Nanopore _Preprocessing_

Before starting any analysis, it is always a good idea to assess the quality of your input data and to discard poor quality base content by trimming and filtering reads.

Generally, we are not interested in the host sequences, rather only those originating from the pathogen itself. It is important to get rid of all host sequences and to only retain sequences that might include a pathogen, both in order to speed up further steps and to avoid host sequences compromising the analysis.

### Input Datasets

- Collection of sequenced Nanopore reads of all samples to be analized in a `fastq or fastq.gz` formate

### Output Datasets

- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` formate

> - Tables indicating total number of reads before and after host sequences trimming, and the host sequences percentages found in each sample, these tables are needed for visualistion performed in **PathoGFAIR Samples Aggregation and Visualisation** workflow.

Reviewer: What is PathGFair? You never mentioned before. I would just remove the mention here

Author: this is just mentioning the name of the workflow, which is mentioned below

## _Taxonomy Profiling_ and Visualisation with Krona

In this workflow, we identify the different organisms found in our samples by assigning taxonomy levels to the reads starting from the kingdom level down to the species level and visualise the result.

It’s important to check what might be the species of a possible pathogen to be found, it gets us closer to the investigation as well as discovering possible multiple pathogenetic infections if any existed.

For taxonomy profiling Kraken2 tool is used along with one of its standard databases available on Galaxy, you can freely choose between Kraken2 different databases based on your input datasets. For visualisation multiple tools can be used, Krona pie chart (as default in this workflow), Phinch interactive tool, Pavian, etc.

> ## _Gene-based Pathogen Identification_

Reviewer: Is it specific for Nanopore data here?

Author: nope its general

Author: Preprocessing for Nanopore, also mention which type of data (metagenomics)

In this workflow we:

1. Perform genome assembly to get contigs, i.e. longer sequences, using metaflye (Kolmogorov et al. 2020) then assembly polishing using medaka consensus pipeline and visualizing the assembly graph using Bandage Image (Wick et al. 2015)
2. Generate reports with AMR genes and VF using ABRicate

As outputs, we get FASTA and Tabular files to track genes and visualise our pathogenic identification through out all samples in **PathoGFAIR Samples Aggregation and Visualisation** workflow.

## Nanopore _Allele-based Pathogen Identification_

Another approach to identifying pathogens is to use an allelic approach by detecting SNPs, i.e. markers showing evolutionary histories of homogeneous strains. This process includes SNP calling, aimed at identifying novel pathogen strains and elucidating discrepancies compared to reference sequences, thereby facilitating the tracking of emerging variants. Within this workflow, both variants and SNPs are discerned, serving as crucial elements for subsequent pathogen identification and variant tracking purposes.

## Pathogen Detection: _PathoGFAIR Samples Aggregation and Visualisation_

> In this workflow, we will aggregate results and use the results from 3 workflows (**_Preprocessing_**, **_Gene-based Pathogen Identification_** and **_Allele-based Pathogen Identification_**) to help tracking pathogenes among samples and visualise all performed analysis by:

Reviewer: ```suggestion
In this workflow, we will aggregate results and use the results from 3 workflows (**_Preprocessing_**, **_Gene-based Pathogen Identification_** and **_Allele-based Pathogen Identification_**) to help track pathogens among samples and visualise all performed analysis by:
```

# Microbiome Workflows

> The following workflows can be used directly for microbiome data analysis, pathogen detection, and tracking purposes. The workflows can also be adapted to any other sequencing technique.

Reviewer 2: I've reviewed this file here, but I've noticed that the readme in the individual directories are duplicates of this text. Could you maybe just very briefly list the workflows here and describe how they relate to each other, then remove the copied text here so we only have to maintain one version of the text ?

To learn more about the following workflows and try them with real datasets, please check out our Microbiome tutorials on the Galaxy Training Network [GTN](https://training.galaxyproject.org/training-material/topics/microbiome/)

> ## Nanopore _Preprocessing_

Reviewer 2: Why italicize just Preprocessing here ? This looks a little odd

Author: I was trying to relate to the workflows' names in the Foodborne paper since the paper I will publish has just the name Preprocesing, not Nanopore Preprocessing. 

But true here it looks very odd, and I cannot really mention the paper here since this directory should be general enough for later to add any microbiome related workflow not just for foodborne

Now I removed all italics

Before starting any analysis, it is always a good idea to assess the quality of your input data and to discard poor-quality base content by trimming and filtering reads.

Generally, we are not interested in the host sequences, but rather only those originating from the pathogen itself. It is important to get rid of all host sequences and to only retain sequences that might include a pathogen, both in order to speed up further steps and to avoid host sequences compromising the analysis.

### Input Datasets

> - Collection of sequenced Nanopore reads of all samples to be analysed in a `fastq or fastq.gz` format.

Reviewer 2: ```suggestion
- Collection of sequenced Nanopore reads of all samples to be analysed in a `fastqsanger` or `fastqsanger.gz` format.
```

### Output Datasets

> - Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` format.

Reviewer 2: ```suggestion
- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format.
```

- Tables indicating total number of reads before and after host sequences trimming, and the host sequences percentages found in each sample.

This workflow is available for trial through our [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)

## _Taxonomy Profiling_ and Visualisation with Krona

In this workflow, we identify the different organisms found in our samples by assigning taxonomy levels to the reads starting from the kingdom level down to the species level and visualise the result.

It’s important to check what might be the species of a possible pathogen to be found, it gets us closer to the investigation as well as discovering possible multiple pathogenetic infections if any existed.

For taxonomy profiling Kraken2 tool is used along with one of its standard databases available on Galaxy, you can freely choose between Kraken2 different databases based on your input datasets. For visualisation multiple tools can be used, Krona pie chart (as default in this workflow), Phinch interactive tool, Pavian, etc.

### Input Datasets

> - Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` format, the output of **Nanopore _Preprocessing_** workflow.

Reviewer 2: ```suggestion
- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format, the output of **Nanopore Preprocessing** workflow.
```

### Output Datasets
- Taxonomy profiling Tabular file, visualisation figures and interactive pie charts.

> This workflow is available for trial through our [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)

Reviewer 2: Not sure what you mean by "for trial" ?

Author: changed to available to try via our ....

> ## _Gene-based Pathogen Identification_

Reviewer 2: ```suggestion
## Gene-based Pathogen Identification
```
I would remove all of the italics, I don't think they add anything here ?

In this workflow, we determine whether the samples are pathogenic or not, by looking for genes known to be linked to pathogenicity or to the pathogenecity character of the organism.

- Virulence Factor (VF): gene products, usually proteins, involved in pathogenicity. By identifying them, we can call a pathogen and its severity level

- Antimicrobial Resistance genes (AMR).

    These type of genes have three fundamental mechanisms of antimicrobial resistance that are enzymatic degradation of antibacterial drugs, alteration of bacterial proteins that are antimicrobial targets, and changes in membrane permeability to antibiotics, which will lead to not altering the target site and spread throughput the pathogenic bacteria decreasing the overall fitness of the host.

In this workflow we:

1. Perform genome assembly to get contigs, i.e. longer sequences, using metaflye (Kolmogorov et al. 2020) then assembly polishing using medaka consensus pipeline and visualizing the assembly graph using Bandage Image (Wick et al. 2015)
2. Generate reports with AMR genes and VF using ABRicate

### Input Datasets

> - Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` format, the output of **Nanopore _Preprocessing_** workflow.

Reviewer 2: ```suggestion
- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format, the output of **Nanopore Preprocessing** workflow.
```

### Output Datasets
- FASTA and Tabular files to track genes and visualise our pathogenic identification through out all samples.

This workflow is available for trial through our [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)

> ## Nanopore _Allele-based Pathogen Identification_

Reviewer 2: ```suggestion
## Nanopore Allele-based Pathogen Identification
```

This workflow identifies pathogens using an allelic approach, detecting Single Nucleotide Polymorphisms (SNPs) to track emerging variants, i.e. markers showing evolutionary histories of homogeneous strains. This process includes SNP calling, aimed at identifying novel pathogen strains and elucidating discrepancies compared to reference sequences, thereby facilitating the tracking of emerging variants. Within this workflow, both variants and SNPs are discerned, serving as crucial elements for subsequent pathogen identification and variant tracking purposes.

To perform the mapping step before variant identification, we used the Minimap2 tool, specifically designed for Nanopore reads. If you're working with Illumina data, you can substitute Minimap2 with Bowtie2.

### Input Datasets

> - Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` format, the output of **Nanopore _Preprocessing_** workflow.

Reviewer 2: ```suggestion
- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastq or fastq.gz` format, the output of **Nanopore Preprocessing** workflow.
```
```suggestion
- Collection of Pre-Processed Sequenced reads of all samples, ready for further analysis with the other workflows, in a `fastqsanger` or `fastqsanger.gz` format, the output of **Nanopore _Preprocessing_** workflow.
```

- A reference genome to the tested pathogen.

### Output Datasets
- VCF files indicating identified variants and SNPs, BAM files with mapping results, and Tabular files with mapping depth and coverage calculations.

This workflow is available for trial through our [GTN tutorial](https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html)

> ## Pathogen Detection: _PathoGFAIR Samples Aggregation and Visualisation_

Reviewer 2: ```suggestion
## Pathogen Detection: PathoGFAIR Samples Aggregation and Visualisation
```

> In this workflow, we will aggregate results and use the results from 3 workflows (**Nanopore _Preprocessing_**, **_Gene-based Pathogen Identification_** and **Nanopore _Allele-based Pathogen Identification_**) to help track pathogens among samples and visualise all performed analysis by:

Reviewer 2: ```suggestion
In this workflow, we will aggregate results and use the results from 3 workflows (**Nanopore Preprocessing**, **Gene-based Pathogen Identification** and **Nanopore Allele-based Pathogen Identification**) to help track pathogens among samples and visualise all performed analysis by:
```
